static/app/views/detectors/edit.tsx

```diff
@@ -1,28 +1,32 @@
 /* eslint-disable no-alert */
-import {Fragment} from 'react';
+import {Fragment, useState} from 'react';
 
 import {Button} from 'sentry/components/core/button';
+import FormModel from 'sentry/components/forms/model';
 import SentryDocumentTitle from 'sentry/components/sentryDocumentTitle';
 import {ActionsProvider} from 'sentry/components/workflowEngine/layout/actions';
 import {BreadcrumbsProvider} from 'sentry/components/workflowEngine/layout/breadcrumbs';
 import EditLayout from 'sentry/components/workflowEngine/layout/edit';
 import {useWorkflowEngineFeatureGate} from 'sentry/components/workflowEngine/useWorkflowEngineFeatureGate';
 import {t} from 'sentry/locale';
 import useOrganization from 'sentry/utils/useOrganization';
+import {MetricDetectorForm} from 'sentry/views/detectors/components/forms/metric';
 import {makeMonitorBasePathname} from 'sentry/views/detectors/pathnames';
 
 export default function DetectorEdit() {
   const organization = useOrganization();
   useWorkflowEngineFeatureGate({redirect: true});
+  const [title, setTitle] = useState(t('Edit Monitor'));
+  const model = new FormModel();
```

Comment:
like the others, not created every render
```suggestion
  const [model] = useState(() => new FormModel());
```